tap_linkedin_ads/streams.py

```diff
@@ -308,126 +313,137 @@ def sync_endpoint(self,
-            #  This function converts unix datetimes, de-nests audit fields,
-            #  tranforms URNs to IDs, tranforms/abstracts variably named fields,
-            #  converts camelCase to snake_case for fieldname keys.
-            # For the Linkedin Ads API, 'elements' is always the root data_key for records.
-            # The data_key identifies the collection of records below the <root> element
-            transformed_data = [] # initialize the record list
-            if self.data_key in data:
-                transformed_data = transform_json(data, self.tap_stream_id)[self.data_key]
-            if not transformed_data or transformed_data is None:
-                LOGGER.info('No transformed_data')
-                break # No data results
-
-            pre_singer_transformed_data = copy.deepcopy(transformed_data)
-            if self.tap_stream_id in selected_streams:
-                # Process records and gets the max_bookmark_value and record_count for the set of records
-                max_bookmark_value, record_count = self.process_records(
-                    catalog=catalog,
-                    records=transformed_data,
-                    time_extracted=time_extracted,
-                    bookmark_field=bookmark_field,
-                    max_bookmark_value=max_bookmark_value,
-                    last_datetime=last_datetime,
-                    parent_id=parent_id)
-                LOGGER.info('%s, records processed: %s', self.tap_stream_id, record_count)
-                total_records = total_records + record_count
-
-            # Loop thru parent batch records for each children objects
-            for child_stream_name in children:
-                if child_stream_name in selected_streams:
-                    # For each parent record
-                    child_obj = STREAMS[child_stream_name]()
-
-                    for record in pre_singer_transformed_data:
-
-                        parent_id = record.get(child_obj.foreign_key)
-
-                        child_stream_params = child_obj.params
-                        # Add children filter params based on parent IDs
-                        if self.tap_stream_id == 'accounts':
-                            account = 'urn:li:sponsoredAccount:{}'.format(parent_id)
-                            owner_id = record.get('reference_organization_id', None)
-                            owner = 'urn:li:organization:{}'.format(owner_id)
-                            if child_stream_name == 'video_ads' and owner_id is not None:
-                                child_stream_params['account'] = account
-                                child_stream_params['owner'] = owner
+        url_list = []
+        if self.tap_stream_id in NEW_PATH_STREAMS:
+            querystring = '&'.join(['%s=%s' % (key, value) for (key, value) in endpoint_params.items()])
+            account_list = config['accounts'].replace(" ", "").split(",")
```

Comment:
This is adds a child stream dependancy in the parent class, `sync_endpoint` method should be modified in the child class.